### ISSUE TITLE
altera-quartus-prime-lite fails to install

### ISSUE THREAD

**Issue by HackerFoo** [bjornfor/nixos-config#5]
From the `pkgs` dir, `nix-env -f . -iA altera-quartuses.altera-quartus-prime-lite` runs into this error on my machine:

```
make[2]: Leaving directory '/build/glibc-2.24/elf'
(sed '/^@/d' include/stubs-prologue.h; LC_ALL=C sort /build/build/csu/stubs /build/build/iconv/stubs /build/build/locale/stubs /build/build/localedata/stubs /build/build/iconvdata/stubs /build/build/assert/stubs /build/build/ctype/stubs /build/build/intl/stubs /build/build/catgets/stubs /build/build/math/stubs /build/build/setjmp/stubs /build/build/signal/stubs /build/build/stdlib/stubs /build/build/stdio-common/stubs /build/build/libio/stubs /build/build/dlfcn/stubs /build/build/malloc/stubs /build/build/string/stubs /build/build/wcsmbs/stubs /build/build/timezone/stubs /build/build/time/stubs /build/build/dirent/stubs /build/build/grp/stubs /build/build/pwd/stubs /build/build/posix/stubs /build/build/io/stubs /build/build/termios/stubs /build/build/resource/stubs /build/build/misc/stubs /build/build/socket/stubs /build/build/sysvipc/stubs /build/build/gmon/stubs /build/build/gnulib/stubs /build/build/wctype/stubs /build/build/manual/stubs /build/build/shadow/stubs /build/build/gshadow/stubs /build/build/po/stubs /build/build/argp/stubs /build/build/crypt/stubs /build/build/nptl/stubs /build/build/rt/stubs /build/build/conform/stubs /build/build/debug/stubs /build/build/mathvec/stubs /build/build/libidn/stubs /build/build/nptl_db/stubs /build/build/inet/stubs /build/build/resolv/stubs /build/build/nss/stubs /build/build/hesiod/stubs /build/build/sunrpc/stubs /build/build/nis/stubs /build/build/nscd/stubs /build/build/streams/stubs /build/build/login/stubs /build/build/elf/stubs) > /build/build/stubs.h
if test -r /nix/store/2qqpwg6fs5635ppgyq7kdjr57y4aj5qs-glibc-2.24-dev/include/gnu/stubs-64.h && cmp -s /build/build/stubs.h /nix/store/2qqpwg6fs5635ppgyq7kdjr57y4aj5qs-glibc-2.24-dev/include/gnu/stubs-64.h; \
then echo 'stubs.h unchanged'; \
else /nix/store/n9acaakxahkv1q3av11l93p7rgd4xqsf-bootstrap-tools/bin/install -c -m 644 /build/build/stubs.h /nix/store/2qqpwg6fs5635ppgyq7kdjr57y4aj5qs-glibc-2.24-dev/include/gnu/stubs-64.h; fi
rm -f /build/build/stubs.h
/build/build/elf/sln /build/build/elf/symlink.list
rm -f /build/build/elf/symlink.list
test ! -x /build/build/elf/ldconfig || LC_ALL=C \
  /build/build/elf/ldconfig  \
                        /nix/store/n42rxmsc4ca65wvjmbw24rr2a50q349p-glibc-2.24/lib /nix/store/n42rxmsc4ca65wvjmbw24rr2a50q349p-glibc-2.24/lib
/build/build/elf/ldconfig: Warning: ignoring configuration file that cannot be opened: /etc/ld.so.conf: No such file or directory
/build/build/elf/ldconfig: Can't create temporary cache file /etc/ld.so.cache~: Permission denied
Makefile:109: recipe for target 'install' failed
make[1]: [install] Error 1 (ignored)
make[1]: Leaving directory '/build/glibc-2.24'
make -r PARALLELMFLAGS="" -C /build/build/../glibc-2.24 objdir=`pwd` localedata/install-locales
make[1]: Entering directory '/build/glibc-2.24'
make  -C localedata install-locales
make[2]: Entering directory '/build/glibc-2.24/localedata'
.././scripts/mkinstalldirs /nix/store/n42rxmsc4ca65wvjmbw24rr2a50q349p-glibc-2.24/lib/locale
mkdir -p -- /nix/store/n42rxmsc4ca65wvjmbw24rr2a50q349p-glibc-2.24/lib/locale
C.UTF-8...cannot open locale definition file `locales/C': No such file or directory
Makefile:205: recipe for target 'install-C.UTF-8/UTF-8' failed
make[2]: *** [install-C.UTF-8/UTF-8] Error 4
make[2]: Leaving directory '/build/glibc-2.24/localedata'
Makefile:421: recipe for target 'localedata/install-locales' failed
make[1]: *** [localedata/install-locales] Error 2
make[1]: Leaving directory '/build/glibc-2.24'
Makefile:9: recipe for target 'localedata/install-locales' failed
make: *** [localedata/install-locales] Error 2
builder for '/nix/store/x0rf51q9lkf54nl75vgkz1vhqwws2bnp-glibc-2.24.drv' failed with exit code 2
cannot build derivation '/nix/store/qqp1yxy3kg6fy230p7y5qrcliviqb9xk-setup-chroot-and-exec.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/xd14wj40rbz5jw44wfhmcc9w3m50z93y-run-in-fhs-env.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/42kc6b9nfhvj1ra3a3dn1nfq52fkhki7-altera-quartus-prime-lite-unwrapped-17.1.1.593.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/a1wmvd8lr3pp47777zvqjsz7pvd4f0wz-altera-quartus-prime-lite-17.1.1.593.drv': 1 dependencies couldn't be built
error: build of '/nix/store/a1wmvd8lr3pp47777zvqjsz7pvd4f0wz-altera-quartus-prime-lite-17.1.1.593.drv' failed
```

`altera-quartus-prime-lite-16` also fails.

nix-info:
```
 - system: `"x86_64-linux"`
 - host os: `Linux 4.19.45, NixOS, 19.09pre180832.c7bcd4277cc (Loris)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.2.2`
 - channels(root): `"nixos-19.09pre180832.c7bcd4277cc"`
 - channels(dusty): `""`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```

**Comment by bjornfor** [comment]
Thanks for the info. I don't have time/energy to fix it now, but I'm pretty sure it works if you use the 18.09 channel instead of unstable (because that's the version I've been using at $dayjob).

**Comment by HackerFoo** [comment]
FYI I tried switching to 19.03, and that didn't work either.
```
 - system: `"x86_64-linux"`
 - host os: `Linux 4.19.46, NixOS, 19.03.172764.50d5d73e22b (Koi)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.2.2`
 - channels(root): `"nixos-19.03.172764.50d5d73e22b"`
 - channels(dusty): `""`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```

**Comment by yorickvP** [comment]
This depends on some glibc overrides that don't work in 19.03, but you can `(import (builtins.fetchTarball  channel:nixos-18.09) ({ config.allowUnfree = true; })).callPackage ./pkgs/altera-quartus.nix` to build this with the 18.09 channel.

**Comment by HackerFoo** [comment]
Thanks, I got it installed:

```
nix-env --set-flag priority 6 tcl
cd pkgs
nix-env --arg pkgs "import (builtins.fetchTarball channel:nixos-18.09) ({ config.allowUnfree = true; })" -f . -iA altera-quartuses.altera-quartus-prime-lite
```

**Comment by yorickvP** [comment]
Maybe https://github.com/TravisWhitaker/nix-quartus will help you?

**Comment by bjornfor** [comment]
Fixed in 532302ea58cab442c3e25bb911a3f9d10c6fcfc9.
